File: app/config/skillpassive.py
# -*- coding: utf-8 -*-

'''
Author: Hannibal
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
'''


skillpassive_map = {};
skillpassive_map[2001] = {"id":2001,"skilldata":[{"lv":1,"effect":"actor['atk']*=1.2","group":1,},{"lv":2,"effect":"actor['atk']*=1.3","group":1,},{"lv":3,"effect":"actor['atk']*=1.4","group":1,},],};
skillpassive_map[2002] = {"id":2002,"skilldata":[{"lv":1,"effect":"actor['spd']+=20","group":1,},{"lv":2,"effect":"actor['spd']+=30","group":1,},],};
skillpassive_map[2003] = {"id":2003,"skilldata":[{"lv":1,"effect":"actor['hpmax']*=1.2","group":1,},],};


class Skillpassive:
	def __init__(self, key):
		config = skillpassive_map.get(key);
		for k, v in config.items():
			setattr(self, k, v);
		return

def create_Skillpassive(key):
		config = skillpassive_map.get(key);
		if not config:
			return
		return Skillpassive(key)

File: tool/table/client/skillpassive.ts
/*
Author: 
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
*/

module config{

let skillpassive_map = null;
export function skillpassive_map_init(config_obj:Object):void{
	skillpassive_map = config_obj["skillpassive_map"];
}


export class Skillpassive extends Object{
	private m_config:any;
	public static m_static_map:Object = new Object();
	constructor(key){
		super();
		this.m_config = skillpassive_map[key];
		for(let i in this.m_config){
			this[i] = this.m_config[i];
		}
		return;
	}
	public static get_Skillpassive(key){
		if(Skillpassive.m_static_map.hasOwnProperty(key) == false){
			Skillpassive.m_static_map[key] = Skillpassive.create_Skillpassive(key);
		}
		return Skillpassive.m_static_map[key];	
	}
	public static create_Skillpassive(key){
		if(skillpassive_map.hasOwnProperty(key)){
			return new Skillpassive(key);
		}
		return null;	
	}
	public static get_cfg_object():Object
	{
		return skillpassive_map;
	}
}
}